### PR TITLE
Update theme colors to match Nautobot branding

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes: #4896
# What's Changed

- Bring in regenerated Bootstrap CSS with updated colors.
- For ease of ongoing development/refinement, add a ~`/theme_preview/`~ `/theme-preview/` URL (only available to authenticated users and only available when DEBUG is enabled) that displays many relevant UI elements (heavily based on the one provided with Bootstrap itself)
   - [x] TODO: document this in the developer docs
- Various minor tweaks and fixes to Nautobot's own CSS files for areas not readily configurable through the base Bootstrap build process.

image

image

For comparison, here's the appearance with the previous CSS files:

image

image

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- n/a Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- n/a Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by glennmatthews** [comment]
Updated screenshots:

image

image

**Comment by glennmatthews** [comment]
Now addressing #4835 as well.

image

image

**Comment by gsnider2195** [comment]
Is the plan to port these changes over to https://github.com/nautobot/nautobot-bootstrap after merging this PR?

**Comment by gsnider2195** [comment]
Not necessarily blocking, just my opinions:

### Dark Mode

#### Not enough contrast in title of panel

image

#### Not enough contrast in links

image

### Light mode

#### Mouseover for tabs too dark

image

#### Active/mouseover badge too dark

image

#### Active/mouseover nav-pills too dark

image

**Comment by gsnider2195** [comment]
I see elements here in the code but they're not rendering in my browser

image

```html
    <div class="col-md-3">
        <h1>Dropdown menus</h1>
        <div class="dropdown theme-dropdown clearfix">
            <a id="dropdownMenu1" href="#" class="sr-only dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Dropdown <span class="caret"></span></a>
            <ul class="dropdown-menu" aria-labelledby="dropdownMenu1">
                <li class="active"><a href="#">Action</a></li>
                <li><a href="#">Another action</a></li>
                <li><a href="#">Something else here</a></li>
                <li role="separator" class="divider"></li>
                <li><a href="#">Separated link</a></li>
            </ul>
        </div>
    </div>
```

File: nautobot/docs/development/core/bootstrap-ui.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Developing the Bootstrap 3 UI
+
+## Introduction
+
+Nautobot's current primary UI is based on [Bootstrap 3](https://getbootstrap.com/docs/3.4/). Since Nautobot 2.1.0, this UI has used a Nautobot-specific custom Bootstrap theme built in the [nautobot-bootstrap](https://github.com/nautobot/nautobot-bootstrap/) GitHub repository and then customized further in the main [nautobot](https://github.com/nautobot/nautobot/) GitHub repository.
+
+## Theme Files
+
+- `nautobot/project-static/bootstrap-3.4.1-dist/css/` - the base Nautobot-themed Bootstrap CSS definitions, directly as compiled from `nautobot-bootstrap`. _These should never be edited manually, only recompiled from `nautobot-bootstrap` and copied as-is into `nautobot`._
```
[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/docs/development/core/bootstrap-ui.md:9]: It's a one-liner: `grunt dist && cp dist/css/bootstrap* ../nautobot/nautobot/project-static/bootstrap-3.4.1-dist/css/`. Probably would make sense to add to the readme of `nautobot-bootstrap`.

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**